### DISCUSSION TITLE
[Feature] Apple TV App

### DISCUSSION THREAD

**Discussion by JesseWebDotCom** [immich-app/immich#3892] [Feature Request]
### The feature

It would be great to view our albums on the Apple TV (and for our family members to do the same).

### Platform

- [ ] Server
- [ ] Web
- [ ] Mobile

**Comment by EricSand** [comment]
+1 for that. Either as a dedicated Apple TV App or (potentially even better and easier) using AirPlay like in the Apple Photos app (not streaming the entire screen but only the picture / video in full quality).
https://support.apple.com/en-kw/guide/iphone/iphd29858fd3/ios#iph97b09249e

**Reply by orywig** [reply]
Yes! I would be just as happy with this and would probably be easier to implement into the iOS app than writing a ATV app from scratch.

**Reply by mertalev** [reply]
This is a smart suggestion! I think this is doable.

**Comment by floco** [comment]
+1 as well, an tvOS app would be great addition to the immich ecosystem. Prefer an app than airplay or chromecast as like this you don't need another device and can browse your photos directly from the comfort of your coach.

**Comment by radityakurnianto** [comment]
Yes, please immich dev, apple tv app would be perfect.

**Reply by hunkyn** [reply]
+1 for this request. For now, I am using Owlfiles which connects to my shared drive and has a screensaver feature to flip photos. It works but a dedicated app for photos would be perfect.

**Comment by ThijsTMK** [comment]
A TV app is the only thing thats sort of keeping me at Synology Photos :( (and I dont want to stay on Synology Photos!)

**Comment by GETandSELECT** [comment]
for me, an Apple TV app is also a killer criterion. 
I really like Immich, I've only been experimenting with it for 3 hours, but it makes a great first impression.

Please don't offer the Apple TV app for free, charge $3 for it so that you can earn money. I'm sure everyone will be happy to pay for it.

**Comment by mertalev** [comment]
Flutter doesn't have good support for this and writing a new client from scratch is not really an option, so it's unlikely this can be added. It could be better suited as a separate project like [ImmichFrame](https://github.com/3rob3/ImmichFrame).

**Comment by fahadshery** [comment]
would love to see Apple TV app...that's the only thing keeping me from using this

**Comment by sidamos** [comment]
+1 for the Apple TV app. +1 for making it a paid app.

**Comment by ruddygarmendez** [comment]
+1 for apple tv app...would be perfect for auto displaying albums during family gatherings

**Comment by CiNcH83** [comment]
Here is how I see it. Not that anybody gives a sh...

Even in 2024, TV's are still cumbersome to navigate. Nobody ever got that right. Neither Samsung or LG, nor Apple, Google or Amazon. Mobile devices on the other hand have become brilliant navigation devices over the years IMHO which is why I am a big time user of Apple's Fotos app together with AirPlay. As soon as those photos are off to the Pi, this is not a thing anymore with Immich unfortunately.

With Apple, Google and Amazon backing [Matter](https://en.wikipedia.org/wiki/Matter_(standard)), there is hope that [Matter Casting](https://matter-smarthome.de/en/know-how/what-is-matter-casting/) will become a thing one day in order to have one common standard for casting to a TV via mobile device.

**Comment by sl33nyc** [comment]
It looks like there's a [community supported version of Flutter](https://medium.com/flutter-community/flutter-for-apple-tv-756fcd5e8113) with an AppleTV target.  The article was written as of Flutter v3.10, but it looks like the developer has continued supporting it as of v3.24, similar to the [current immich version](https://github.com/immich-app/immich/blob/main/mobile/.fvmrc#L2).

I recognize that building against a community supported fork of the Flutter SDK isn't ideal, but something is better than nothing?  Also, combining that with a $3 fee on the Apple Store app would hopefully offset any, hopefully minor tweaks to build using the Flutter fork.

**Reply by CiNcH83** [reply]
I believe that a TV photos app will always be bad due to the way a TV is being controlled. Casting to me is the only viable solution as a mobile device is perfectly suited for navigating a huge library with the TV only acting as a big screen for dispaying photos.

**Reply by mertalev** [reply]
I think casting is a much smaller lift. Making the iOS app start up in Apple TV is difficult on its own, but it's also only the first step: designing the UI to work well for a TV would require a ton of work to the point that you may as well write a dedicated native Swift app.

**Comment by anst84** [comment]
For all people having iPhones and AppleTVs who doesn't want to pay a lot of money to apple recurringly (or other reasons) for extending their iCloud storage, the support for an Immich App running on Apple TVOS would be huge. And, people with such setup should be really common given the Apple footprint, and thus the demand should be high out there. 

Just to give an example from my family. We are a family of five all having iPhones, iPads and three AppleTVs. When we want to browse family photos we just click the Photos app on the AppleTV. One click away - super simple. If you compare this to casting it is just not that simple. My non-techy family members will always choose the easier option. The only way I could persuade them to change would be to install the Immich app on the Apple TV (dream scenario) and (re)move the Photos app down to the bottom of the screen.

This said, I have a great respect for the challenges and efforts required to accomplish this. I also want to add that I'm extremely satisfied with the Immich app and what it provides as of now.

**Comment by neponn** [comment]
I would also like to see an Apple TV app - right now I’m looking for something that might work. I’m shifting my photography workflow to HDR, complete with new Mac and OLED tv. The one thing that’s missing is an easy way to do an HDR slide show on the TV.

**Comment by luxychris2009** [comment]
YES!  +1 FOR ME!  APPLE TV APP IS CRUCIAL!  The only thing I see missing from this scenario.  Thanks Dev Team!  :)
